src/prefect/agent.py

```diff
@@ -94,31 +91,31 @@ async def get_and_submit_flow_runs(self) -> List[FlowRun]:
             seconds=self.prefetch_seconds or PREFECT_AGENT_PREFETCH_SECONDS.value()
         )
 
-        # Use the work queue id or load one from the name
-        work_queue_id = self.work_queue_id or await self.work_queue_id_from_name()
-        if not work_queue_id:
-            return []
+        submittable_runs = []
 
-        try:
-            submittable_runs = await self.client.get_runs_in_work_queue(
-                id=work_queue_id, limit=10, scheduled_before=before
-            )
-        except httpx.HTTPStatusError as exc:
-            if exc.response.status_code == status.HTTP_404_NOT_FOUND:
-                raise ValueError(
-                    f"No work queue found with id '{work_queue_id}'"
-                ) from None
-            else:
-                raise
+        # load runs from each work queue
+        for work_queue in await self.get_work_queues():
 
-        # Check for a paused work queue for display purposes
-        if not submittable_runs:
-            work_queue = await self.client.read_work_queue(work_queue_id)
+            # print a nice message if the work queue is paused
             if work_queue.is_paused:
```

Comment:
Since we're inspecting the queue here we may want to `yield` each work queue from `get_work_queues()` to ensure that data is fresh.